sunpy/database/commands.py

```diff
@@ -94,8 +96,20 @@ def __call__(self):
             operation()
 
     def undo(self):
+        # To prevent errors in SQLa 2.0 we have to first add all removed entries back in before we add any ~database.Tags.
+        for operation in self._operations:
+            if isinstance(operation, RemoveEntry):
+                # we also filter out ~database.commands.RemoveEntry where type is not ~database.tables.DatabaseEntry to do later.
+                if isinstance(operation.entry, DatabaseEntry):
+                    operation.undo()
         for operation in self._operations:
-            operation.undo()
+            if isinstance(operation, RemoveEntry):
+                if not isinstance(operation.entry, DatabaseEntry):
+                    # undo all ~database.commands.RemoveEntry where type is not ~database.tables.DatabaseEntry
+                    operation.undo()
+            elif not isinstance(operation, RemoveEntry):
```

Comment:
This should be equivalent, although maybe it's more readable as is?
```suggestion
            else:
```